addons/wowl/static/src/components/webclient/webclient.ts

```diff
@@ -1,10 +1,127 @@
 import { ActionContainer } from "../../services/action_manager/action_manager";
-import { Component } from "@odoo/owl";
+import { Component, hooks } from "@odoo/owl";
 import { NavBar } from "../navbar/navbar";
-import { OdooEnv } from "../../types";
+import { ActionContext, OdooEnv } from "../../types";
+import { useService } from "../../core/hooks";
 
 export class WebClient extends Component<{}, OdooEnv> {
   static components = { ActionContainer, NavBar };
   static template = "wowl.WebClient";
+  menus = useService("menus");
+  action_manager = useService("action_manager");
+  router = useService("router");
   Components = this.env.registries.Components.getEntries();
+  mainAction: any = {};
```

Comment:
havin to use this is annoying, and only serves in loadRouterStates to determine if we should make a switchView instead.
And in Push state to retrieve the action's info and push them to the url